lib/Test/Stream/Formatter/TAP.pm

```diff
@@ -11,10 +11,33 @@ sub OUT_STD()  { 0 }
 sub OUT_ERR()  { 1 }
 sub OUT_TODO() { 2 }
 
+use Carp qw/croak/;
+
 use Test::Stream::Exporter qw/import exports/;
 exports qw/OUT_STD OUT_ERR OUT_TODO/;
 no Test::Stream::Exporter;
 
+my %CONVERTERS = (
+    'Test::Stream::Event::Ok'        => \&_ok_event,
+    'Test::Stream::Event::Skip'      => \&_skip_event,
+    'Test::Stream::Event::Note'      => \&_note_event,
+    'Test::Stream::Event::Diag'      => \&_diag_event,
+    'Test::Stream::Event::Bail'      => \&_bail_event,
+    'Test::Stream::Event::Exception' => \&_exception_event,
+    'Test::Stream::Event::Subtest'   => \&_subtest_event,
+    'Test::Stream::Event::Plan'      => \&_plan_event,
+);
+
+sub register_event {
+    my $class = shift;
+    my ($type, $convert) = @_;
+    croak "Event type is a required argument" unless $type;
+    croak "Event type '$type' already registered" if $CONVERTERS{$type};
+    croak "The second argument to register_event() must be a code reference"
+        unless $convert && ref($convert) eq 'CODE';
+    $CONVERTERS{$type} = $convert;
+}
+
```

Comment:
Q) Is it expected nobody will wish to redefine how a TAP event is rendered? 

I somehow get the impression somebody *may* wish to inject their own custom formatter for an existing event, but this may not be the right place for this.

If so: Comments to that effect should accompany the source for source divers, and POD documentation for register_event should be blindingly clear about this.